hw/ip/ascon/doc/background.md

```diff
@@ -0,0 +1,41 @@
+## Background
+
+This section provides simple diagrams on Sponge, Ascon, round based implementations, and the round function. It describes the basic principles of Sponge and Ascon. It also lists some performance figures from different (existing) implementations.
+
+
+#### Duplex Sponge
+
+A cryptographic public permutation is a cryptographic primitive that efficiently applies a permutation over b-bit blocks.  The description of the permutation itself is fully public (unlike block ciphers that take a secret input) and the security stems from how the permutation is used. The duplex sponge construction[^2] is a popular method for implementing keyed primitives, such as Authenticated Encryption with Associated Data (AEAD)[^3] from public permutations. For a full description of the duplex construction and its applications we refer to[^4]. Figure 1 depicts a simplified version of an AEAD using the duplex construction. The state of the algorithm is initialized by concatenating the secret key K and a nonce N, then applying the cryptographic permutation. Then, the state is divided into a r-bit rate and a c-bit capacity. The associated data A and the input M are divided into blocks and the blocks are XORed with the rate. The input corresponds to the plaintext P for encryption and to the ciphertext C for decryption. To avoid certain attacks while making the scheme slightly more efficient, the different stages (initialization, processing of the associated data, processing of the message) must be separated by a value a and m added to the capacity c. This is known as Monkey Sponge[^5]. After each block is XORed, the permutation is applied. After all the blocks have been absorbed, the permutation is applied again and the t-bit authentication tag is extracted. In Figure 1, the tag consists of one block, so size of tag t=r.
```

Comment:
```suggestion
A cryptographic public permutation is a cryptographic primitive that efficiently applies a permutation over b-bit blocks. The description of the permutation itself is fully public (unlike block ciphers that take a secret input) and the security stems from how the permutation is used. The duplex sponge construction[^2] is a popular method for implementing keyed primitives, such as Authenticated Encryption with Associated Data (AEAD)[^3] from public permutations. For a full description of the duplex construction and its applications we refer to[^4]. Figure 1 depicts a simplified version of an AEAD using the duplex construction. The state of the algorithm is initialized by concatenating the secret key K and a nonce N, then applying the cryptographic permutation. Then, the state is divided into a r-bit rate and a c-bit capacity. The associated data A and the input M are divided into blocks and the blocks are XORed with the rate. The input corresponds to the plaintext P for encryption and to the ciphertext C for decryption. To avoid certain attacks while making the scheme slightly more efficient, the different stages (initialization, processing of the associated data, processing of the message) must be separated by a value a and m added to the capacity c. This is known as Monkey Sponge[^5]. After each block is XORed, the permutation is applied. After all the blocks have been absorbed, the permutation is applied again and the t-bit authentication tag is extracted. In Figure 1, the tag consists of one block, so size of tag t=r.
```